test/functional/wallet_migration.py

```diff
@@ -32,6 +33,13 @@
     generate_keypair,
 )
 
+SQLITE3_IMPORTED = False
+try:
+    import sqlite3 # type: ignore[import]
+    SQLITE3_IMPORTED = True
+except ImportError:
+    pass
```

Comment:
In 3854d6b56c50b26f7b61dfb37f5230726d561cf9 "tests: Check that the last hardened cache upgrade occurs"

I think I have found a way of getting rid of this kind of duplicated try-import-except-setfileconstant instances in multiple files by introducing a `inspect_database` function in the test framework. Mentioned in overall review comment.

I think it would be quite nice to have this inspect function but I don't feel like blocking the PR only for this.